internal/xds/translator/listener.go

```diff
@@ -180,8 +180,9 @@ func buildXdsTCPListener(
 		},
 	}
 
-	if ipFamily != nil && *ipFamily == egv1a1.DualStack {
+	if ipFamily != nil && (*ipFamily == egv1a1.DualStack || *ipFamily == egv1a1.IPv6) {
```

Comment:
will this make it impossible to generate a pure IPv6 listener?

Reply:
To my understanding, this flag allows envoy to respond to IPv4 requests when the listening address is `::`. A pure IPv6 listener on a single stack IPv6 cluster should work, it seems the E2E tests confirm this though I haven't tested it personally.

Is a pure IPv6 listener possible on a dual stack cluster with this change? No. The alternative would be to introduce another parameter to the EnvoyProxy spec which allows this behavior.